cpp/src/io/comp/gpuinflate.cu

```diff
@@ -1220,5 +1224,41 @@ void gpu_copy_uncompressed_blocks(device_span<device_span<uint8_t const> const>
   }
 }
 
+void decompress_check(device_span<decompress_status> stats,
+                      bool* d_any_block_failure,
+                      rmm::cuda_stream_view stream)
+{
+  if (stats.empty()) { return; }  // Early exit for empty stats
+
+  auto block_failures = rmm::device_uvector<bool>(stats.size(), stream);
+  thrust::transform(rmm::exec_policy(stream),
+                    stats.begin(),
+                    stats.end(),
+                    block_failures.begin(),
+                    [] __device__(auto const& stat) { return stat.status != 0; });
+
+  // Allocate temporary storage
+  size_t storage_bytes = 0;
+  cub::DeviceReduce::Reduce(nullptr,
+                            storage_bytes,
+                            block_failures.data(),
+                            d_any_block_failure,
+                            block_failures.size(),
+                            thrust::logical_or{},
+                            false,
+                            stream.value());
+  auto temp_storage = rmm::device_buffer{storage_bytes, stream};
+
+  // Run reduction
+  cub::DeviceReduce::Reduce(temp_storage.data(),
```

Comment:
I see that `cub::` functions are used in many places. However, using them typically requires two passes as here. Is there any way to refactor that out into helper functions so we can call and use by a shorter way?
```
void/auto cub_reduce(...) {
  size_t storage_bytes = 0;
  cub::DeviceReduce::Reduce(...);
  auto temp_storage = rmm::device_buffer{storage_bytes, stream};
  cub::DeviceReduce::Reduce(...);
}
```
Then at caller site, just one LOC:
```
cub_reduce(...);
```